x-pack/plugins/session_view/public/components/Process/index.tsx

```diff
@@ -0,0 +1,295 @@
+      height: 100%;
+      pointer-events: none;
+      content: '';
+      margin-left: -${depth * TREE_INDENT}px;
+      border-left: 4px solid ${borderColor};
+      background-color: ${bgColor};
+      width: calc(100% + ${depth * TREE_INDENT}px);
+    }
+  `;
+
+  const wrapperCSS = `
+    position: relative;
+    padding-left: 8px;
+    vertical-align: middle;
+    color: ${euiTheme.colors.mediumShade};
+    word-break: break-all;
+    min-height: 24px;
+    line-height: 24px;
+  `;
+
+  function renderSessionLeader() {
+    const { name, user } = process.getLatest().process;
+    const sessionIcon = interactive ? 'consoleApp' : 'compute';
+
+    return (
+      <>
+        <EuiIcon type={sessionIcon} /> <b css={darkTextCSS}>{name}</b>
+        &nbsp;
+        <FormattedMessage id="kbn.sessionView.startedBy" defaultMessage="started by" />
+        &nbsp;
+        <EuiIcon type="user" /> <b css={darkTextCSS}>{user.name}</b>
+      </>
+    );
+  }
+
+  // TODO: not customizable for now (cmd previously offered a template string to render)
+  const template = () => {
+    const {
+      args,
+      working_directory: workingDirectory,
+      exit_code: exitCode,
+    } = process.getLatest().process;
+    const { searchMatched } = process;
+
+    if (searchMatched !== null) {
+      const regex = new RegExp(searchMatched);
+
+      // TODO: should we allow some form of customization via settings?
+      let text = `${workingDirectory} ${args.join(' ')}`;
+
+      text = text.replace(regex, (match) => {
+        return `<span style="${searchHighlightCSS}">${match}</span>`;
+      });
+
+      return (
+        <>
+          {/* eslint-disable-next-line react/no-danger */}
+          <span dangerouslySetInnerHTML={{ __html: text }} />
+        </>
+      );
+    }
+
+    const workingDirCSS = `
+      color: ${euiTheme.colors.successText};
+    `;
+
+    return (
+      <>
+        <span css={workingDirCSS}>{workingDirectory}</span>&nbsp;
+        <span css={darkTextCSS}>{args[0]}</span>&nbsp;
+        {args.slice(1).join(' ')}
+        {exitCode && <small> [exit_code: {exitCode}]</small>}
+      </>
+    );
+  };
+
+  function renderProcess() {
+    const userEnteredIconCSS = `
+      position: absolute;
+      width: 9px;
+      height: 9px;
+      margin-left: -11px;
+      margin-top: 8px;
+    `;
+
+    return (
+      <>
+        {process.isUserEntered() && <EuiIcon css={userEnteredIconCSS} type="user" />}
+        <EuiIcon type="console" /> {template()}
+      </>
+    );
+  }
+
+  /**
+   * calls up to terminal to show details for this command
+   */
+  function onProcessClicked(e: MouseEvent): void {
+    e.stopPropagation();
+
+    const selection = window.getSelection();
+
+    // do not select the command if the user was just selecting text for copy.
+    if (selection && selection.type === 'Range') {
+      return;
+    }
+
+    onProcessSelected(process);
+  }
+
+  const id = process.getEntityID();
+
+  return (
+    <>
```

Comment:
nitpick: not a huge fan of returning a react fragment as the root element, since it becomes harder to style and inspect later on